modules/javafx.media/src/main/native/gstreamer/plugins/av/videodecoder.c

```diff
@@ -295,9 +424,129 @@ static void videodecoder_state_reset(VideoDecoder *decoder)
+        if (!decoder->sws_getContext_func)
+        {
+            gst_element_message_full(GST_ELEMENT(decoder), GST_MESSAGE_ERROR,
+                    JFX_GST_ERROR, JFX_GST_INVALID_LIBSWSCALE,
+                    g_strdup("Error: Failed to find \"sws_getContext()\" in libswscale"), NULL,
+                    ("videodecoder.c"), ("videodecoder_init_converter"), 0);
+            return FALSE;
+        }
+
+        decoder->sws_freeContext_func = dlsym(decoder->swscale_module, "sws_freeContext");
+        if (!decoder->sws_freeContext_func)
+        {
+            gst_element_message_full(GST_ELEMENT(decoder), GST_MESSAGE_ERROR,
+                    JFX_GST_ERROR, JFX_GST_INVALID_LIBSWSCALE,
+                    g_strdup("Error: Failed to find \"sws_freeContext()\" in libswscale"), NULL,
+                    ("videodecoder.c"), ("videodecoder_init_converter"), 0);
+            return FALSE;
+        }
+
+        decoder->sws_scale_func = dlsym(decoder->swscale_module, "sws_scale");
+        if (!decoder->sws_scale_func)
+        {
+            gst_element_message_full(GST_ELEMENT(decoder), GST_MESSAGE_ERROR,
+                    JFX_GST_ERROR, JFX_GST_INVALID_LIBSWSCALE,
+                    g_strdup("Error: Failed to find \"sws_scale()\" in libswscale"), NULL,
+                    ("videodecoder.c"), ("videodecoder_init_converter"), 0);
+            return FALSE;
+        }
+    }
+
+    if (decoder->dest_frame)
+    {
+        av_frame_free(&decoder->dest_frame);
+        decoder->dest_frame = NULL;
+    }
+
+    if (decoder->sws_context)
+    {
+        decoder->sws_freeContext_func(decoder->sws_context);
+        decoder->sws_context = NULL;
+    }
+
+    decoder->sws_context =
+            decoder->sws_getContext_func(decoder->width, decoder->height,
+                                         base->frame->format, decoder->width,
+                                         decoder->height, AV_PIX_FMT_YUV420P,
+                                         SWS_BILINEAR, NULL, NULL, NULL);
+
+    if (decoder->sws_context == NULL)
+        return FALSE;
+
+    decoder->dest_frame = av_frame_alloc();
+    if (decoder->dest_frame == NULL)
+        return FALSE;
+
+    decoder->dest_frame->format = AV_PIX_FMT_YUV420P;
+    decoder->dest_frame->width  = decoder->width;
+    decoder->dest_frame->height = decoder->height;
+    int ret = av_frame_get_buffer(decoder->dest_frame, 32);
+    if (ret < 0)
+    {
+        av_frame_free(&decoder->dest_frame);
+        decoder->dest_frame = NULL;
+        decoder->sws_freeContext_func(decoder->sws_context);
+        decoder->sws_context = NULL;
+        return FALSE;
+    }
+
+    return TRUE;
+}
+
+static gboolean videodecoder_convert_frame(VideoDecoder *decoder)
+{
+    BaseDecoder *base = BASEDECODER(decoder);
+
+    if (decoder->sws_context == NULL || decoder->dest_frame == NULL ||
+            decoder->sws_scale_func == NULL)
+        return TRUE;
```

Comment:
Not really. videodecoder_init_converter() should fail if we need converter, but did not able to initialize it. videodecoder_convert_frame() return TRUE here, since it assumes that it will never be called if we need converter and fail to initialize it. So, TRUE here means we do not need to convert frame. I modified this logic by checking base->frame->format value to make it clear.